packages/blade/src/components/Input/_decisions/decisions.md

```diff
@@ -0,0 +1,212 @@
+# Input Decisions <!-- omit in toc -->
+
+- [Design](#design)
+- [BaseInputField API](#baseinputfield-api)
+- [TextInput API](#textinput-api)
+  - [Web & React Native Attributes for showing keyboard types and autocomplete suggestions based on input](#web--react-native-attributes-for-showing-keyboard-types-and-autocomplete-suggestions-based-on-input)
+  - [Return Key types for web and Native](#return-key-types-for-web-and-native)
+- [PasswordInput](#passwordinput)
+- [Notes](#notes)
+- [Open questions](#open-questions)
+- [References](#references)
+
+## Design
+
+- [TextInput](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=10953%3A191574)
+- [PasswordInput](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=10953%3A180381)
+- [OTPInput]()
+- [CardInput]()
+- [TextArea]()
+
+## BaseInputField API
+
+## TextInput API
+
+**Sample Usage**
+
+```tsx
+<TextInput
+  label="Enter Name"
+  labelPosition="left"
+  placeholder="Enter your name"
+  leadingIcon={UserIcon}
```

Comment:
+1 if it has to be only icons.

But should the content only be limited to icons? I think there can be many use cases where consumers would want to have text instead. Could be `@gmail.com`, `Rs.`, telephone codes as `+91` etc. I think in that case we should have props `trailingContent` & `leadingContent` that can accept `Icon` or `string`.

Reply:
update based on design: we decided to only accept `icon` as a prop and always fix it to left because that's the general design decision we want to impose. right side is always preserved for showing only actions(show password, clear button) on and suffix content like `.00` or `@gmail.com`

>I think in that case we should have props trailingContent & leadingContent that can accept Icon or string.

yes we do have `prefix` and `suffix` props already listed in the table. this is just a minimalistic example for demo purpose